Marlin/src/HAL/hal_gpio.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Snapmaker2-Modules Firmware
+ * Copyright (C) 2019-2020 Snapmaker [https://github.com/Snapmaker]
+ *
+ * This file is part of Snapmaker2-Modules
+ * (see https://github.com/Snapmaker/Snapmaker2-Modules)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "hal_gpio.h"
+
+const static GPIO_TypeDef* PORTs[] = {
+  GPIOA,
+  GPIOB,
+  GPIOC,
+  GPIOD,
+  GPIOE,
+  GPIOF,
+  GPIOG,
+};
+
+const static uint32_t RCCTable[] = {
+  RCC_APB2Periph_GPIOA,
+  RCC_APB2Periph_GPIOB,
+  RCC_APB2Periph_GPIOC,
+  RCC_APB2Periph_GPIOD,
+  RCC_APB2Periph_GPIOE,
+  RCC_APB2Periph_GPIOF,
+  RCC_APB2Periph_GPIOG
+};
+
+void GpioRemap() {
+  GPIO_PinRemapConfig(GPIO_Remap_SWJ_JTAGDisable, ENABLE);
+}
+
+void GpioInit(uint8_t Port, GPIOMode_TypeDef Mode) {
```

Comment:
Firstly, the  formal parameter 'Port', it actually indicates the 'Pin' number here，so should name it to 'pin'.
Secondly, per the code style, the  name of formal parameter should start with lower case